### DISCUSSION TITLE
View the data in hive json format locally, how to add jar, where to add

### DISCUSSION THREAD

**Discussion by feipenghou** [dbeaver/dbeaver#10813] [Q&A] [answered]
SQL 错误 [40000] [42000]: Error while compiling statement: FAILED: RuntimeException MetaException(message:java.lang.ClassNotFoundException Class org.apache.hive.hcatalog.data.JsonSerDe not found)

**Comment by uselessscat** [comment, marked as answer]
Hi @feipenghou. Sadly i don't know chinese, so im using translator. 
Some communities defines how properly make questions, contributions, pull requests, etc. I think here, [they defined](https://github.com/dbeaver/dbeaver#notes) you have to use [Discussions](https://github.com/dbeaver/dbeaver/discussions) to make questions.

I don't know what you are tring to do, but i will try to answer your question.
In the documentation, says that after creating a connection, Dbeaver downloads the required Jars. You can read here https://github.com/dbeaver/dbeaver/wiki/Apache-Hive. I tested that and a windows appears when you click the tab "Driver Properties".
![image](https://user-images.githubusercontent.com/15332878/103255611-b2324f00-4968-11eb-8555-9cd5981677de.png)

Here is the windows
![image](https://user-images.githubusercontent.com/15332878/103255715-210fa800-4969-11eb-8838-3e42eb952ba4.png)

If using that method doesn't work i would try to setup drivers manually.
To get the driver, i think you need the classes from hadoop and hive, [look here](https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients#HiveServer2Clients-RunningtheJDBCSampleCode). Specifically the driver jar, I think maybe some one of these https://mvnrepository.com/artifact/org.apache.hive/hive-jdbc
Then you have to configure Dbeaver to use it, [here are more details](https://github.com/dbeaver/dbeaver/wiki/Database-drivers) of how to do it. Basically you have to add the jars as a library in the driver configuration.

After that you would connect sucessfull, if not you can provide more details of what are you trying.
You can use platforms like https://stackoverflow.com/ to ask for help too. There Dbeaver have [a tag](https://stackoverflow.com/questions/tagged/dbeaver)
hope it helps
